server/app/services/geo/AddressSuggestion.java

```diff
@@ -0,0 +1,45 @@
+package services.geo;
+
+import com.google.auto.value.AutoValue;
+import services.Address;
+
+/**
+ * Represents an address candidate in the context of address candidates
+ *
+ * <p>See {@link AddressCandidates} for details on address candidates
+ */
+@AutoValue
+public abstract class AddressSuggestion {
+  public static Builder builder() {
+    return new AutoValue_AddressSuggestion.Builder();
+  }
+  /** returns an address as a single line */
+  public abstract String getSingleLineAddress();
+
+  /** returns the location object, which conatiins x and y coordinates */
+  public abstract AddressLocation getLocation();
+
+  /**
+   * returns an integer which represents how well the candidate address matches with the user
+   * inputted address
+   *
+   * <p>score has the range [0-100] sorted in descending order
```

Comment:
suggestion (blocking): the score itself is a scalar so not sorted. State what a score means: is a low score or a high score indicative of a high-confidence suggestion?